Source/Blazorise/BarIcon.razor

```diff
@@ -1,13 +1,15 @@
 ﻿@inherits BaseComponent
-@if (!HasCustomRegistration)
+@if ( !HasCustomRegistration )
```

Comment:
I'm thinking if `BarIcon `is needed as a separate component. Maybe it's good to implement `@( IconName as RenderFragment )` on an existing `Icon`? 

That would leave us with custom `b-bar-icon` class, and how to detect it needs to be used. Maybe see if Icon is placed inside `BarItem`?

Reply:
I agree! My only issues is that the it needs to apply this class inside any `Bar` entity (`BarItem`, `BarLink`, `BarDropdownToggle`, etc).

As far as I'm aware, `[CascadingParameter]` only allows to flow to immediate children? (So I couldn't have `[CascadingParameter] BarItem` because of this).

Do you know of any other way I can achieve this? 😅

Reply:
Well, for every BarLink, BarDropdownToggle, etc. there is always placed in at least one BarItem. So I think it doesn't matter how many cascaded BarItem's there is as long as there is one, the classname will be applied.

Reply:
Or just it needs to know that it is placed inside of `Bar`, and that's it. What do you think?

Reply:
Do we transfer this question to another PR with other issues?

Reply:
> Or just it needs to know that it is placed inside of Bar, and that's it. What do you think?

I like this idea, just not sure how to do it other than cascading down everything 🤔

Reply:
Maybe do the `<CascadingValue Value="Bar" Name="Bar">` this one time? 😅

Reply:
Actually, just having `[CascadingParameter] protected Bar ParentBar { get; set; }` is enough. All we need to know it's not null.